packages/bcp/src/cosmwasmconnection.ts

```diff
@@ -185,8 +188,34 @@ export class CosmWasmConnection implements BlockchainConnection {
     }
   }
 
-  public watchAccount(_account: AccountQuery): Stream<Account | undefined> {
-    throw new Error("not implemented");
+  public watchAccount(query: AccountQuery): Stream<Account | undefined> {
```

Comment:
This is good.

I have some questions for when I try to integrate with the frontend, but not pressing.

Basically, if I create the stream and subscribe on when mounting a component, how do I close it properly on leaving? Looking at https://github.com/staltz/xstream#subscribe I would guess something like:

```js
const sub = client.watchAccount({address: address}).subscribe({ /* callbacks */});
// when done
sub.cancel() // ???? this is not on the README
```

Reply:
Also, this is only on bcp level.
I think it would be nice to have on sdk level as well (or done there and then the bcp is a wrapper around it).
I query balance in simple react app to put in the header, but it never updates unless you reload. This `watchAccount` function would be a perfect match to the `getAccount` just added to `@cosmwasm/sdk`

Reply:
First point: right. And then there is `Subscription.unsubscribe()`.

Second point: All this streaming stuff barely works. It is a mess. You can choose between HTTP polling and websockets that break and don't reconnect. I think it is safe to call this road a failure. There is nothing the library can offer to the app which the app can't do on its own.